front-end/src/components/Workspace.js

```diff
@@ -19,15 +18,30 @@ class Workspace extends React.Component {
         this.model = new DiagramModel();
         this.engine.setModel(this.model);
         this.engine.setMaxNumberPointsPerLink(0);
+        this.state = {nodes: []};
```

Comment:
Is there a default set of nodes from the backend?

Reply:
See the `componentDidMount` method. The constructor usually just sets the shape of the state data, and it's populated elsewhere.